src/video/wayland/SDL_waylandevents.c

```diff
@@ -661,6 +662,30 @@ keyboard_handle_keymap(void *data, struct wl_keyboard *keyboard,
         input->xkb.keymap = NULL;
         return;
     }
+
+    /*
+     * See https://blogs.s-osg.org/compose-key-support-weston/
+     * for further explanation.
+     */
+
+    /* Look up the preferred locale, falling back to "C" as default */
+    if (!(locale = getenv("LC_ALL")))
+        if (!(locale = getenv("LC_CTYPE")))
+            if (!(locale = getenv("LANG")))
+                locale = "C";
```

Comment:
We should probably replace this with SDL_GetLocale()

Reply:
I'm not sure how to convert between `SDL_Locale` to the locale string format expected by the xkb functions. I checked the x11 driver code and it seems to just directly use the relevant environment variables.

Reply:
Ah, fair enough then - in that case, let's just replace `getenv` with `SDL_getenv` for stdlib compatibility and leave it at that.